pdp/static/js/pdp_raster_map.js

```diff
@@ -1,223 +1,115 @@
-        var maxTimeIndex, units, startDate, layerTime;
-
-        maxTimeIndex = ddsToTimeIndex(maxTime[0]);
-        unitsSince = dasToUnitsSince(unitsSince[0]);
-        units = unitsSince[0];
-        startDate = unitsSince[1];
-        calendar = unitsSince[2];
-        layerTime = new CfTime(units, startDate, calendar);
-        layerTime.setMaxTimeByIndex(maxTimeIndex);
-        ncwms_layer.times = layerTime; // Future access through ncwmslayer?
-	// Some pages don't have date pickers
-	if ($(".datepickerstart").length > 0) {
-            setTimeAvailable(layerTime.sDate, layerTime.eDate);
-	}
+        var indexCount = ddsToTimeIndex(maxTime[0]);
+        var cfTimeSystem = dasToCfTimeSystem(unitsSince[0], indexCount);
+        ncwms_layer.cfTimeSystem = cfTimeSystem;
+
+        // If page has datepicker(s), set them up.
+        if ($(".datepickerstart").length > 0) {
+            setTimeAvailable(cfTimeSystem);
+	    }
     });
 }
 
-function setTimeAvailable(begin, end) {
-    //TODO: only present times available in ncwms capabilities for this layer
-    var yearRange = begin.getFullYear().toString(10) + ":" + end.getFullYear().toString(10);
+function transferDate(date, newSystem, fallbackDate) {
+    // Transfer `date`, which is a `CfDatetime` object with an
+    // associated `CfTimeSystem` to the (new) `cfTimeSystem`, if possible.
+    //
+    // Transfer is possible if `date`:
+    //  (a) exists
+    //  (b) contains a datetime (year, month, day, etc.) compatible
+    //      with `cfTimeSystem`, meaning the datetime does not throw an
+    //      error when a new `CfDatetime` is created using its values,
+    //      which in turn means it is compatible with `cfTimeSystem.calendar`
+    //      and does not exceed the index bounds of `cfTimeSystem`.
+    //
+    // Otherwise return `fallbackDate`.
+    if (!date) {
+        return fallbackDate;
+    }
+    if (_.isEqual(date.system, newSystem)) {
+        // Same CfTimeSystem, therefore compatible, no need to create new
+        // CfDatetime.
+        return date;
+    }
+    try {
+        var dt = date.toCalendarDatetime().datetime;
+        var result = new calendars.CfDatetime.fromDatetime(
+            newSystem, dt.year, dt.month, dt.day
+        );
+        return result;
+    } catch(error) {
+        return fallbackDate;
+    }
+}
+
+function setTimeAvailable(cfTimeSystem) {
+    // We associate, using jQuery's `.data()` method, a date value
+    // to the start and end datepicker elements. This value encodes the meaning
+    // of the date entered in each element as a `CfDatetime`, which carries
+    // all information about the CF time system (units, start date,
+    // calendar, index count) as well as the value of the date proper.
+    //
+    // We attempt to transfer the existing dates to the next time system.
+    //
+    // Note: we don't have yearRange any more.
+    // TODO: Find out what yearRange was for and compensate.
```

Comment:
I think `yearRange` was a UI parameter that told the calendar-style datepicker which years should be present in its dropdown menu. You've providing that functionality via checking date validity, so I don't think we need to worry about any missing functionality from `yearRange`.